lms/djangoapps/lti_provider/tests/test_outcomes.py

```diff
@@ -363,3 +299,128 @@ def test_response_with_failing_status_field(self):
             major_code='<imsx_codeMajor>failure</imsx_codeMajor>'
         )
         self.assertFalse(outcomes.check_replace_result_response(response))
+
+
+class TestAssignmentsForProblem(ModuleStoreTestCase):
+    """
+    Test cases for the assignments_for_problem method in outcomes.py
+    """
+    def setUp(self):
+        super(TestAssignmentsForProblem, self).setUp()
+        self.course = CourseFactory.create()
+        self.course_key = CourseLocator(
+            org='some_org',
+            course='some_course',
+            run='some_run'
+        )
+        self.chapter = ItemFactory.create(parent=self.course, category="chapter")
+        self.vertical = ItemFactory.create(parent=self.chapter, category="vertical")
+        self.unit = ItemFactory.create(parent=self.vertical, category="unit")
```

Comment:
Please use bulk operations for this. So:

```python
self.store.bulk_operations(self.course_key, emit_signals=False):
    self.chapter = ItemFactory.create(parent=self.course, category="chapter")
    self.vertical = ItemFactory.create(parent=self.chapter, category="vertical")
    self.unit = ItemFactory.create(parent=self.vertical, category="unit")
```

Or the `cls.*` equivalent if you change this to a `SharedModuleStoreTestCase`. Using bulk operations lowers the number of Mongo roundtrips, which can drastically affect test run speed.